ui/src/components/FormFields.tsx

```diff
@@ -29,10 +34,26 @@ export const Label = styled.label`
   font-weight: bold;
 `;
 
-export const TextInput = styled.input.attrs({ type: "text" })`
+interface TextInputProps {
+  error?: boolean;
+}
+
+const inputError = (error: boolean) =>
+  error &&
+  css`
+    &,
+    &:focus {
+      border-color: var(--color-red);
+    }
+  `;
+
+export const TextInput = styled.input.attrs<TextInputProps>((props) => ({
+  type: props.type || "text",
+}))<TextInputProps>`
   ${inputBorder};
   ${inputFont};
   ${inputLayout};
+  ${({ error }) => error && inputError(error)}
```

Comment:
I think error is unnecessary since if error is false inputError returns false. 

On a sidenote: wow i have never seen this style before is it common practice?

Reply:
Not sure if I understood you but you're right there's some redundancy. I've refactored inputError to not be a function and just return the error style, since the `error &&` is already a guard in place.